src/unread/unreadSelectors.js

```diff
@@ -257,9 +258,8 @@ export const getUnreadCountForNarrow: Selector<number, Narrow> = createSelector(
 
       pm: (emails, ids) => {
         if (ids.length > 1) {
-          // TODO this should go somewhere central like recipient.js
-          const userIds = [...ids, ownUserId].sort((a, b) => a - b).join(',');
-          const unread = unreadHuddles.find(x => x.user_ids_string === userIds);
+          const unreadsKey = pmUnreadsKeyFromPmKeyIds(ids, ownUserId);
+          const unread = unreadHuddles.find(x => x.user_ids_string === unreadsKey);
           return unread ? unread.unread_message_ids.length : 0;
         } else {
           const senderId = ids[0];
```

Comment:
Could call `pmUnreadsKeyFromPmKeyIds` to get `senderId` here, I suppose—as long as `senderId` and `unreadPms[].sender_id` are made to agree in being a `number` or a `string` (for the `.find` that they're used in here).

Reply:
Yeah. I think that will most naturally come after converting these to our own `Immutable.Map` or similar data structure, and perhaps after converting the both of these (`state.unread.pms` and `state.unread.huddles`) into a single such data structure.

With the data structures as they are, with `.sender_id` a number, if we ran `ids` through `pmUnreadsKeyFromPmKeyIds` we'd have to convert it back from string to number in order to do the search (or else convert every `sender_id` into a string to do the comparison, which would be inefficient.) That conversion would make sense only because we know the key isn't just any string but has a particular structure... but the same facts about its structure mean we can just use the number directly and skip converting to a string and back.

Reply:
> Yeah. I think that will most naturally come after converting these to our own `Immutable.Map` or similar data structure, and perhaps after converting the both of these (`state.unread.pms` and `state.unread.huddles`) into a single such data structure.

Makes sense—I figured that was in the plan. 🙂